server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java

```diff
@@ -194,12 +198,8 @@ Set<String> sourcePaths(String field) {
         }
 
         String resolvedField = field;
-        int lastDotIndex = field.lastIndexOf('.');
-        if (lastDotIndex > 0) {
-            String parentField = field.substring(0, lastDotIndex);
-            if (fullNameToFieldType.containsKey(parentField)) {
-                resolvedField = parentField;
-            }
```

Comment:
It will be so nice to get rid of this, I think it's more or less the last place where we do this annoying dot parsing.